### PR TITLE
document new uuid functions

### PR DESCRIPTION
**Pull request by ephys**
This PR documents PR https://github.com/sequelize/sequelize/pull/16864, and removes mentions of the legacy functions

### CODE REVIEW

File: docs/models/data-types.mdx [hunk 1]
```diff
@@ -222,18 +226,36 @@ For UUIDs, use `DataTypes.UUID`. It becomes the `UUID` data type for PostgreSQL
...
 ```
 
-<JsDefaultCaution />
+In supported dialects, Sequelize will set the default value to the appropriate function, as shown in the table below.
+These dialects can also use these two functions in migrations.
+
+In all other dialects, Sequelize will generate the UUID value itself, in JavaScript.
+This means that they cannot use these two functions in migrations.
+
+<UuidSupportTable />
```
Comment (WikiRik) [inline docs/models/data-types.mdx:246]: Do we want to add footnotes for postgres <13 and mysql <8.0.31?

File: docs/querying/raw-queries.mdx [hunk 2]
```diff
@@ -483,11 +485,32 @@ Attributes support a shorthand syntax for casting. See [Casting syntax in `sql.a
...
+
+These functions are supported by the following dialects:
+
+```sql
+-- postgres
+INSERT INTO users (id) VALUES (gen_random_uuid())
+
+-- mssql
+INSERT INTO users (id) VALUES (NEWID())
+```
```
Comment (WikiRik) [inline docs/querying/raw-queries.mdx:505]: Isn't this included in the table already?

File: docs/models/data-types.mdx (hunk 1 above)
Comment (ephys) [reply docs/models/data-types.mdx:246]: Added info for postgres < 13 here, and added a note for mysql < 8.0.13 in the defining-models page

File: docs/querying/raw-queries.mdx (hunk 2 above)
Comment (ephys) [reply docs/querying/raw-queries.mdx]: Reworked

**Review by WikiRik — APPROVED** [review]

**Pull request merged**